backend/src/main/java/com/google/sps/data/Participant.java

```diff
@@ -1,65 +1,86 @@
 package com.google.sps.data;
 
+import com.google.common.base.MoreObjects;
 import com.google.common.base.Preconditions;
-import java.time.ZonedDateTime;
 
-/** A user who want to be matched. */
+/** A user who wants to be matched. */
 public final class Participant {
 
-  /** Datastore ID */
-  private final long id;
   /** Google username (ldap) */
   private final String username;
-  /** Time user is starts being available */
-  private final ZonedDateTime startTimeAvailable;
+  /** Time user starts being available */
+  private final long startTimeAvailable;
   /** Time user is available until */
-  private final ZonedDateTime endTimeAvailable;
+  private final long endTimeAvailable;
   /** How long user wants to chat */
   private final int duration;
+  /** Id of match in datastore, 0 if never found a match (can assign 0 at construction) */
+  private final long matchId;
   /** Time of submitted form */
   private final long timestamp;
 
   /** Initialize constructor fields */
   public Participant(
-      long id,
       String username,
-      ZonedDateTime startTimeAvailable,
-      ZonedDateTime endTimeAvailable,
+      long startTimeAvailable,
+      long endTimeAvailable,
       int duration,
+      long matchId,
       long timestamp) {
-    this.id = id;
     this.username = username;
-
     Preconditions.checkArgument(
-        startTimeAvailable.isBefore(endTimeAvailable),
+        startTimeAvailable < endTimeAvailable,
         "Start available time must be before end available time.");
     this.startTimeAvailable = startTimeAvailable;
     this.endTimeAvailable = endTimeAvailable;
     this.duration = duration;
+    this.matchId = matchId;
     this.timestamp = timestamp;
   }
 
-  public long getId() {
-    return id;
-  }
-
   public String getUsername() {
     return username;
   }
 
-  public ZonedDateTime getStartTimeAvailable() {
+  public long getStartTimeAvailable() {
     return startTimeAvailable;
   }
 
-  public ZonedDateTime getEndTimeAvailable() {
+  public long getEndTimeAvailable() {
     return endTimeAvailable;
   }
 
   public int getDuration() {
     return duration;
   }
 
+  public long getMatchId() {
+    return matchId;
+  }
+
   public long getTimestamp() {
     return timestamp;
   }
+
+  /** Return participant with new matchId and nulled out availability */
+  public Participant foundMatch(long newMatchId) {
+    return new Participant(
+        username,
+        /* startTimeAvailable= */ 0,
+        /* endTimeAvailable= */ 0,
+        /* duration= */ 0,
```

Comment:
If I understand correctly, the idea is to zero all of these fields out so that the matching algorithm will ignore them due to low values? (Zero minute meeting at midnight on January 1, 1970 😄) 

We can do this in a followup PR but I recommend replacing this with an explicit value keeping track of the matched state. I would recommend using an Enum, even if it ends up just having two values for LOOKING_FOR_MATCH and MATCHED. (As for why to avoid a boolean, my old team used to have an expression: Enums get extended, booleans get deprecated) Then we don't need to depend on implementation details of the matching algorithm to ensure matched users are filtered out. 

WDYT? Fix here or file an issue for the next PR?

Reply:
Good point, I can fix it here. Enums aren't valid datastore property types, though, so should I just do a boolean?

Reply:
Enums can be converted to and from ints

Reply:
Created MatchStatus enum